### PR TITLE
First part of MVP-pipeline

### PR DESCRIPTION
**Pull request by MrKahr**
Be sure to test pipeline thoroughly

### CODE REVIEW

File: modules/config/config_enums.py
Comment (TheRealMorgenfrue) [inline modules/config/config_enums.py:1]: L. 54: Change class name to better explain what it's doing (e.g. FeatureSelectCriterion)

File: modules/config/config_template_gen.py
Comment (TheRealMorgenfrue) [inline modules/config/config_template_gen.py:1]: Remove unnecessary comment on lines:
  - 34
  - 62
  - 71
  - 73
  - 97

Line 54:  Non-obvious abbreviation. Consider more descriptive name.

File: modules/dataPreprocessing/cleaner.py
Comment (TheRealMorgenfrue) [inline modules/dataPreprocessing/cleaner.py:1]: General: 
  - Do we write `return self.df.copy(deep=True)` or `return self.getDataFrame()`? 

Line 180: Non-obvious abbreviation. Consider more descriptive name.
Line 187: Do we need logging to tell it's done? Consider removing as we'll know where it fails, if it fails.

File: modules/dataPreprocessing/outlier_analysis/outlier_processor.py
Comment (TheRealMorgenfrue) [inline modules/dataPreprocessing/outlier_analysis/outlier_processor.py:1]: General:
  - Do we write `return self.df.copy(deep=True)` or `return self.getDataFrame()`? 
  - Variable `outliers` unused?

Line 85: Do we need logging to tell it's done? Consider removing as we'll know where it fails, if it fails.
Line 86: Should be a deep copy?

File: modules/dataPreprocessing/transformer.py
Comment (TheRealMorgenfrue) [inline modules/dataPreprocessing/transformer.py:1]: Lines 45-47: Remove. We do not write `Returns` in docstrings if we do not return anything.
Line 110: Comment out or remove method `matrixDistance` if not used.
Line 141: Non-standard method name. Should be camelCase.
Line 175: Non-standard method name. Should be camelCase.
Line 214: Missing type annotation.
Line 268: Do we need logging to tell it's done? Consider removing as we'll know where it fails, if it fails.

File: modules/modelSelection/model_selection.py
Comment (TheRealMorgenfrue) [inline modules/modelSelection/model_selection.py:1]: Line 72: Do we need logging to tell it's done? Consider removing as we'll know where it fails, if it fails.

File: modules/dataPreprocessing/feature_selector.py
Comment (TheRealMorgenfrue) [inline modules/dataPreprocessing/feature_selector.py:1]: General: 
  - Consider making `self.X` and `self.y` private as they should only be directly accessed within this class.
  - Remove `X` and `y` as parameters for **ALL** functions and replace with `self.X` and `self.y`.
  - Rename `self.X` and `self.y` to something more descriptive (e.g. `self._train_x` and `self._target_y`).

Discuss with reviewer:
  1. Consider not calling `chi2` etc. methods directly but use them through `GenericUnivariateSelect`.
  2. We should prolly refactor the method `genericUnivariateSelect` and change the template to accommodate `1` (and to get rid of `x_labels`).
  3.  Do we write `return self.df.copy(deep=True)` or `return self.getDataFrame()`? 

Line 9: Remove unused import.
Line 47: Missing type annotation. Rename `x` and `y` to something more descriptive (e.g. `train_x` and `target_y`).
Line 186: Remove line.
Line 292: Remove method. Is used in training.
Line 296: Remove method. Is used in training.
Line 300: Consider making private if it's only used within this class.

File: modules/modelTesting/model_testing.py
Comment (TheRealMorgenfrue) [inline modules/modelTesting/model_testing.py:1]: General:
  - The first parameter for all `@classmethod` must be `cls`, not `self`.

Line 71: Do we need logging to tell it's done? Consider removing as we'll know where it fails, if it fails.
Line 73: Non-standard method name. Should be camelCase.

**Review by TheRealMorgenfrue — CHANGES_REQUESTED** [review]

File: modules/dataPreprocessing/cleaner.py
Comment (MrKahr) [reply modules/dataPreprocessing/cleaner.py:1]: 1) We currently consider returning a deep copy the safest option. First option is currently standard. (Discuss)
2) Logging when the data cleaner is done frames logging messages to their respective parts making them easier to understand when output in terminal.

File: modules/dataPreprocessing/outlier_analysis/outlier_processor.py
Comment (MrKahr) [reply modules/dataPreprocessing/outlier_analysis/outlier_processor.py:1]: We currently consider returning a deep copy the safest option. First option is currently standard. (Discuss)
 
 Logging when the data cleaner is done frames logging messages to their respective parts making them easier to understand when output in terminal.

File: modules/dataPreprocessing/transformer.py
Comment (MrKahr) [reply modules/dataPreprocessing/transformer.py:1]: Logging when the data cleaner is done frames logging messages to their respective parts making them easier to understand when output in terminal.

File: modules/modelSelection/model_selection.py
Comment (MrKahr) [reply modules/modelSelection/model_selection.py:1]: Logging when the data cleaner is done frames logging messages to their respective parts making them easier to understand when output in terminal.

**Review by TheRealMorgenfrue — APPROVED** [review]

**Pull request merged**